crates/fluvio/src/consumer/retry.rs

```diff
@@ -0,0 +1,279 @@
+
+            // Determine the offset for reconnection.
+            let offset = if let Some(next) = inner.next_offset_to_read {
+                match Offset::absolute(next) {
+                    Ok(off) => off,
+                    Err(e) => {
+                        warn!(target: SPAN_RETRY, "Error creating offset: {}", e);
+                        return Some(Err(ErrorCode::OffsetOutOfRange));
+                    }
+                }
+            } else {
+                inner.consumer_config.offset_start.clone()
+            };
+
+            // Update the consumer configuration with the new offset.
+            let mut new_config = inner.consumer_config.clone();
+            new_config.offset_start = offset;
+
+            // Reconnect loop: keep trying until a new stream is created.
+            loop {
+                info!(target: SPAN_RETRY, "Reconnecting to stream");
+                match fluvio_client
+                    .consumer_with_config_inner(new_config.clone())
+                    .await
+                {
+                    Ok(new_stream) => {
+                        backoff.reset();
+                        current_stream = Box::pin(new_stream) as BoxConsumerStream;
+                        info!(
+                            target: SPAN_RETRY,
+                            "Created new consume stream with offset: {:?}", inner.next_offset_to_read
+                        );
+                        break;
+                    }
+                    Err(e) => {
+                        backoff_and_wait(&mut backoff).await;
+                        attempts += 1;
+                        warn!(
+                            target: SPAN_RETRY,
+                            "Could not connect to stream on {}: {}",
+                            inner.consumer_config.topic,
+                            e
+                        );
+
+                        match inner.consumer_config.retry_mode {
+                            RetryMode::TryUntil(max) if attempts >= max => {
+                                return Some(Err(ErrorCode::MaxRetryReached));
+                            }
+                            RetryMode::Disabled => {
+                                return Some(Err(ErrorCode::Other(format!("{}", e))));
+                            }
+                            _ => {} // Continue retrying.
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Creates an exponential backoff configuration.
+fn create_backoff() -> ExponentialBackoff {
+    ExponentialBackoffBuilder::default()
+        .factor(BACKOFF_FACTOR)
+        .min(BACKOFF_MIN_DURATION)
+        .max(BACKOFF_MAX_DURATION)
+        .build()
+        .expect("Failed to build backoff configuration")
+}
+
+/// Waits for the duration determined by the exponential backoff.
+async fn backoff_and_wait(backoff: &mut ExponentialBackoff) {
+    let wait_duration = backoff.wait();
+    debug!(
```

Comment:
let's change to info so this is important log

Reply:
fixed!